### PR TITLE
Scherer

### PR DESCRIPTION
**Pull request by schererc**
- Now able to calculate tabulated dihedral interactions in force matching with and without restrained qrsolve
- contains routines necessary for nonbonded 3-body interactions in force matching

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -28,6 +28,10 @@ enable_language(CXX)
 #####################################################################
 
 include(CheckCXXCompilerFlag)
+check_cxx_compiler_flag("-std=c++11" COMPILER_SUPPORTS_CXX11)
+if(COMPILER_SUPPORTS_CXX11)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=gnu++0x")
```
Comment (junghans) [inline CMakeLists.txt]: That is a bit strange. You check for `-std=c++11` and then you inject `-std=gnu++0x` ? And why do you want to add `-Wall`? - That has nothing to do with c++11!

File: include/votca/csg/beadtriple.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/* 
+ * Copyright 2009-2011 The VOTCA Development Team (http://www.votca.org)
```
Comment (junghans) [inline include/votca/csg/beadtriple.h]: Should be 2016!

File: include/votca/csg/triplelist.h [hunk 3]
```diff
@@ -0,0 +1,100 @@
+/* 
+ * Copyright 2009-2011 The VOTCA Development Team (http://www.votca.org)
```
Comment (junghans) [inline include/votca/csg/triplelist.h]: 2016?

File: netbeans/csg_fmatch/Makefile [hunk 4]
```diff
@@ -1,92 +0,0 @@
-#
```
Comment (junghans) [inline netbeans/csg_fmatch/Makefile:1]: Why?

File: share/scripts/inverse/table_switch_border.pl [hunk 5]
```diff
@@ -0,0 +1,70 @@
+#! /usr/bin/perl -w
+#
+# Copyright 2009-2011 The VOTCA Development Team (http://www.votca.org)
```
Comment (junghans) [inline share/scripts/inverse/table_switch_border.pl]: 2016!

File: src/tools/csg_fmatch.cc [hunk 6]
```diff
@@ -269,11 +283,16 @@ void CGForceMatching::EvalConfiguration(Topology *conf, Topology *conf_atom)
...
-        for(int i=0; i<conf->BeadCount(); ++i) {
+        for(int i=0; i<conf->BeadCount(); ++i) {                
+//            cout << "conf->getBead(" << i << ")->getPos(): " << conf->getBead(i)->getPos() << endl;                
+//            cout << "_top_force->getBead(" << i << ")->getPos(): " << _top_force.getBead(i)->getPos() << endl; 
             conf->getBead(i)->F() -= _top_force.getBead(i)->getF();
             vec d = conf->getBead(i)->getPos() - _top_force.getBead(i)->getPos();
-            if(abs(d) > 1e-6)
-                throw std::runtime_error("One or more bead positions in mapped and reference force trajectory differ by more than 1e-6");
+//            cout << "vec d of bead " << i << ": " << d << "abs(d): " << abs(d) << endl; 
+            if(abs(d) > 1e-5)//altered to 1e-5, otherwise it can be a too strict criterion
```
Comment (junghans) [inline src/tools/csg_fmatch.cc:292]: I think we should make 1e-5, a parameter, which is read from the xml file and use 1e-5 the default!

Comment (junghans) [inline src/tools/csg_fmatch.cc:292]:
```
double dist = options->get("fmatch.dist").as<double>();
```

**Comment by junghans** [comment]
Two general comments:
- Copyright headers need update
- Why did you remove the netbeans Makefiles?

**Comment by junghans** [comment]
Build failed due to https://travis-ci.org/votca/csg/jobs/156518043#L4324

```
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Failed
```

The cmake check should require C++11!

File: CMakeLists.txt (hunk 1 above)
Comment (schererc) [reply CMakeLists.txt]: To be honest, I also do not understand why this options have been added. I tried 

set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11") instead and it works.

File: include/votca/csg/beadtriple.h (hunk 2 above)
Comment (schererc) [reply include/votca/csg/beadtriple.h]: Sorry, I copied beadpair.h and edited it. There the copyright was 2009-2011 as well

File: include/votca/csg/triplelist.h (hunk 3 above)
Comment (schererc) [reply include/votca/csg/triplelist.h]: Right!

File: share/scripts/inverse/table_switch_border.pl (hunk 5 above)
Comment (schererc) [reply share/scripts/inverse/table_switch_border.pl]: yes!

File: netbeans/csg_fmatch/Makefile (hunk 4 above)
Comment (schererc) [reply netbeans/csg_fmatch/Makefile:1]: I deleated the whole netbeans folder. Probably that was not a good idea.

File: src/tools/csg_fmatch.cc (hunk 6 above)
Comment (schererc) [reply src/tools/csg_fmatch.cc:292]: Ok. That would be a good idea. That is right.

File: CMakeLists.txt (hunk 1 above)
Comment (junghans) [reply CMakeLists.txt]: Well you have to do a check with `check_cxx_compiler_flag()` as C++11 is only supported in >=gcc-4.8 and hence VOTCA won't compile with older compilers.

**Comment by junghans** [comment]
@MrTheodor, @symashayak : please have a look, too!

@schererc: please add that `fmatch.dist` option as discussed above and document the new xml option [here](https://github.com/votca/csg/blob/master/share/xml/csg_defaults.xml.in#L447) (for per-interaction options) or [here](https://github.com/votca/csg/blob/master/share/xml/csg_defaults.xml.in#L3) (for global options).

**Comment by junghans** [comment]
Comments addressed in 9b46a942f01bbe3ea406a5b155b7e168a0facce0.

**Comment by schererc** [comment]
Hi Christoph,

ich habe jetzt die option dist an beiden Stellen in der /share/xml/
csg_defaults.xml.in eingefuegt und auch csg_fmatch so abgeaendert, dass
jetzt dist eingelesen wird. Des weiteren habe ich die Option periodic in
/share/xml/csg_defaults.xml eingefuegt (Zeile 422).

Vermutlich sollte man das auch noch im Manual aendern?

Wenn ich die Option dist nicht im settings.xml file angebe, dann meckert er
bei mir zumindest jetzt (Ich habe das mal ausprobiert). Ich weiss nicht, ob
das mit der Default-Option so geklappt hat.

VG und Vielen Dank

Christoph

2016-09-05 17:48 GMT+02:00 Christoph Junghans notifications@github.com:

**Comment by junghans** [comment]
Default values only work in `csg_inverse`, you will need to add a `options->exists()` call around it.

**Comment by junghans** [comment]
The manual is always up to date here: http://doc.votca.org/manual.pdf

**Comment by schererc** [comment]
Hi, ich habe jetzt den _options.exists call implementiert und es scheint zu
funktionieren.

Ok, das manual scheint up-to-date zu sein. Ich habe aber die flag periodic
(fuer bonded interactions bei FM) irgendwie nicht gefunden. Bei der
Beschreibung der iOptionen im settings.xml file habe ich aber irgendwie eh
nur den non-bonded und nicht den bonded Teil gefunden.

VG

2016-09-06 18:23 GMT+02:00 Christoph Junghans notifications@github.com:

**Comment by junghans** [comment]
I see `cg.bonded.periodic` in the 1st block of section 10.4 of the manual, so it seems ok.

**Pull request merged**